tempesta_fw/sock.c

```diff
@@ -1049,7 +1130,14 @@ ss_connect(struct sock *sk, struct sockaddr *uaddr, int uaddr_len, int flags)
 	if (sk->sk_state != TCP_CLOSE)
 		return -EISCONN;
 
-	return sk->sk_prot->connect(sk, uaddr, uaddr_len);
+	if (ss_active_guard_enter(SS_SOFTIRQ_VAL))
```

Comment:
Logically, this should be `SS_LISTEN_VAL` as this is the connection establishing phase, and the connection is not live.